src/toil/batchSystems/slurm.py

```diff
@@ -156,6 +165,32 @@ def getJobExitCode(self, batchJobID):
             elif exit_reason == BatchJobExitReason.FINISHED:
                 pass #self._collectMetrics(slurm_job_id)
             return exit_code
+
+        def getJobExitCodes(self, batchJobIDs):
+            """
+            Get job exit codes for given list of batch job IDs.
+            :param batchJobIDs: list of strings of the form "<job>[.<task>]".
+            :return: list of integer job exit code.
+            """
+
+            slurm_job_ids = [batchJobID.split('.')[0] for batchJobID in batchJobIDs]
+            status_dict = self._get_job_details(slurm_job_ids)
+
+            batch_dict = {}
+            for batchJobID in batchJobIDs:
```

Comment:
what if we run one single query to get job status of all jobs, and parse them?
```console
$ sacct -X -j 14334784 --format JobID,State,ExitCode -P
JobID|State|ExitCode
14334784_1|RUNNING|0:0
14334784_2|RUNNING|0:0
14334784_3|RUNNING|0:0
...
14334784_55|RUNNING|0:0
14334784_56|RUNNING|0:0
14334784_57|FAILED|127:0
```

Reply:
Ah but the rate-limit is with `slumrctld` and `sacct` uses `slurmdb`. Ok so no big issue then.

Reply:
You are actually right -- `sacct` is also rate limited in the current setup (HPC team said). But that's exactly what this function is doing, `status_dict = self._get_job_details(slurm_job_ids)` creates a dictionary of job statuses using s single query! 😄